### PR TITLE
add opaque closure fixes to allow running on 1.8/1.9

### LINKED ISSUE #75: `opaque_closure_method: invalid syntax`
on master:
```julia
julia> ∇((x,y) -> x*y)(1,2)
ERROR: opaque_closure_method: invalid syntax
Stacktrace:
 [1] (::Diffractor.∂⃖{1})(::var"#3#4", ::Int64, ::Vararg{Int64})
   @ Diffractor ~/.julia/dev/Diffractor/src/stage1/generated.jl:215
 [2] Diffractor.∂⃖(::Function, ::Vararg{Any})
   @ Diffractor ~/.julia/dev/Diffractor/src/interface.jl:25
 [3] (::∇{var"#3#4"})(::Int64, ::Vararg{Int64})
   @ Diffractor ~/.julia/dev/Diffractor/src/interface.jl:121
 [4] top-level scope
   @ REPL[19]:1

julia> versioninfo()
Julia Version 1.9.0-DEV.470
Commit 902a5c199d (2022-05-05 15:00 UTC)
```

### PR DESCRIPTION
**Pull request by oscardssmith**
with help from @keno

[Edit: fixes #75 ]

### CODE REVIEW

File: src/stage1/recurse.jl [hunk 1]
```diff
@@ -852,7 +854,7 @@ function transform!(ci, meth, nargs, sparams, N)
     non_dce_finish!(compact)
     ir = complete(compact)
     ir = compact!(ir)
-    Core.Compiler.verify_ir(ir)
+    #Core.Compiler.verify_ir(ir)
```
Comment (Keno) [inline src/stage1/recurse.jl:857]: Should add an `allow_frontend_forms` argument to verify_ir that allows :opaque_closure_method in value position.

**Comment by Keno** [comment]
Add CI for 1.8 first?

**Comment by Keno** [comment]
Fixes #75

File: src/stage1/recurse.jl (hunk 1 above)
Comment (oscardssmith) [reply src/stage1/recurse.jl:857]: cf https://github.com/JuliaLang/julia/pull/45786

Comment (mcabbott) [reply src/stage1/recurse.jl:885]: Something like this?
```suggestion
    if VERSION > v"1.9.0-DEV.854"
        Core.Compiler.verify_ir(ir; allow_frontend_forms=true)
    end
```

Comment (oscardssmith) [reply src/stage1/recurse.jl]: not quite, but close.

**Comment by oscardssmith** [comment]
I believe this should now work on 1.7, 1.8, and 1.9. I had to disable a few tests, but all the tests I disabled were broken on all julia versions I tried, so I don't think they were ever working.

**Review by mcabbott — APPROVED** [review]
Can't claim to understand every line, but locally this works, 1.9 & 1.7.

**Comment by oscardssmith** [comment]
I think this is ready to merge. There are more issues, but this is a good set.

**Pull request merged**